### PR TITLE
feat: Nam - CRUD product feature

### PR DESCRIPTION
**Pull request by huynamboz**

### CODE REVIEW

File: src/Core/Api/V1/Controllers/Product/productController.cs [hunk 1]
```diff
@@ -0,0 +1,57 @@
+﻿using e_commerce_server.src.Core.Modules.Product.Dto;
+using e_commerce_server.src.Core.Modules.Product.Service;
+using e_commerce_server.Src.Core.Database.Data;
+using e_commerce_server.Src.Packages.HttpException;
+using Microsoft.AspNetCore.Http;
+using Microsoft.AspNetCore.Mvc;
+
+namespace e_commerce_server.src.Core.Api.V1.Controllers.Product
+{
+    [Route("api/[controller]")]
```
Comment (phamhongphuc1403) [inline src/Core/Api/V1/Controllers/Product/productController.cs]: should be api/v1/[controller]

File: src/Core/Api/V1/Controllers/Product/productController.cs [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+    [ApiController]
+    public class productController : ControllerBase
```
Comment (phamhongphuc1403) [inline src/Core/Api/V1/Controllers/Product/productController.cs]: product should be plural

File: src/Core/Api/V1/Controllers/Product/productController.cs (hunk 1 above)
Comment (huynamboz) [reply src/Core/Api/V1/Controllers/Product/productController.cs]: Dạ

File: src/Core/Database/Data/ProductData.cs [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+        [Required]
+        public int discount { get; set; }
+        [Required]
+        public DateTime created_at { get; set; }
+        [Required]
+        public DateTime updated_at { get; set;}
+        [Required]
+        public string thumbnail_url { get; set; }
+        [Required]
+        public string product_status { get; set; }
```
Comment (phamhongphuc1403) [inline src/Core/Database/Data/ProductData.cs:30]: it should be int type not string

File: src/Core/Database/Data/ProductData.cs [hunk 4]
```diff
@@ -0,0 +1,42 @@
...
+        public int price { get; set; }
+       
+        [Required]
+        public int discount { get; set; }
+        [Required]
+        public DateTime created_at { get; set; }
+        [Required]
+        public DateTime updated_at { get; set;}
+        [Required]
+        public string thumbnail_url { get; set; }
```
Comment (phamhongphuc1403) [inline src/Core/Database/Data/ProductData.cs]: it should be on another table with one-to-many relationship

File: src/Core/Modules/Product/Service/ProductService.cs [hunk 5]
```diff
@@ -0,0 +1,101 @@
...
+
+namespace e_commerce_server.src.Core.Modules.Product.Service
+{
+    public class ProductService
+    {
+        readonly MyDbContext _dbContext;
+        public ProductService(MyDbContext context) {
+            _dbContext= context;
+        }
+        public object editProductByID(ProductDto productDto, int idProduct)
```
Comment (phamhongphuc1403) [inline src/Core/Modules/Product/Service/ProductService.cs]: where's the authorization check?

File: src/Core/Modules/Product/Service/ProductService.cs [hunk 6]
```diff
@@ -0,0 +1,101 @@
...
+        {
+            var product = _dbContext.Products.FirstOrDefault(p => p.id== idProduct);
```
Comment (phamhongphuc1403) [inline src/Core/Modules/Product/Service/ProductService.cs]: should add try-catch exception

File: src/Core/Modules/Product/Service/ProductService.cs [hunk 7]
```diff
@@ -0,0 +1,101 @@
...
+            if (product == null)
+            {
+                throw new BadRequestException("Product not exist");
```
Comment (phamhongphuc1403) [inline src/Core/Modules/Product/Service/ProductService.cs]: should use ProductEnum

File: src/Core/Modules/Product/Service/ProductService.cs [hunk 8]
```diff
@@ -0,0 +1,101 @@
...
+            {
+                product.description = productDto.description;
+                product.price = productDto.price;
+                product.name = productDto.name;
+                product.updated_at = DateTime.Now;
+                product.category_id = productDto.category_id;
+                product.keyword = productDto.keyword;
+                product.discount = productDto.discount;
+                product.thumbnail_url = productDto.thumbnail_url;
+                _dbContext.SaveChanges();
```
Comment (phamhongphuc1403) [inline src/Core/Modules/Product/Service/ProductService.cs]: should use try-catch

File: src/Core/Modules/Product/Service/ProductService.cs [hunk 9]
```diff
@@ -0,0 +1,101 @@
...
+                    keyword = productDto.keyword,
+                    user_id = idUser,
+                    category_id = productDto.category_id,
+                };
+                _dbContext.Add(newproduct);
+                _dbContext.SaveChanges();
+                return productDto;
+            } catch (Exception ex)
+            {
+                return new BadRequestException(ex.ToString());
```
Comment (phamhongphuc1403) [inline src/Core/Modules/Product/Service/ProductService.cs]: should be an internal exception

File: src/Core/Modules/Product/Service/ProductService.cs [hunk 10]
```diff
@@ -0,0 +1,101 @@
...
+            ).ToList();
+
+                return new
+                {
+                    success = true,
+                    data = listProduct
+                };
+            } catch (Exception ex)
+            {
+                return ex;
```
Comment (phamhongphuc1403) [inline src/Core/Modules/Product/Service/ProductService.cs]: what kind of exception?

**Pull request merged**